src/js/thirdparty/undici.ts

```diff
@@ -265,6 +285,63 @@ function Undici() {
 
 class Dispatcher extends EventEmitter {}
 class Agent extends Dispatcher {}
+
+type ProxyAgentOptions = {
+  uri: string;
+  /**
+   * @deprecated use opts.token
+   */
+  auth?: string;
+  token?: string;
+  headers?: IncomingHttpHeaders;
+};
+
+class ProxyAgent extends Dispatcher {
+  #uri: string;
+  #token?: string;
+  // #headers?: Dict<string | string[]>;
+
+  constructor(uri: string, options: ProxyAgentOptions) {
+    super();
+
+    if (!uri || typeof uri !== "string") throw new TypeError("uri must be a string");
+    if (uri.startsWith("https://")) throw new TypeError("proxy over TLS not implemented");
+    if (!uri.startsWith("http://")) throw new Error("uri must be a valid URL string with the protocol `http://`");
+
+    let token;
+    if (options) {
+      if (typeof options !== "object") throw new TypeError("options must be an object");
+      // Auth token can either be passed via `token` or `auth` prop, though `auth` is now deprecated
+      const { token: _token, auth } = options;
+      token = _token ?? auth;
+      if (token && (typeof token !== "string" || !token.startsWith("Basic ")))
+        throw new TypeError("token must be a string formated as `Basic <TOKEN_IN_BASE_64>`");
+    }
+
+    this.#uri = uri;
+    this.#token = token;
+    // this.#headers = options.headers;
+  }
+
+  get [kBunUndiciProxyAgentUri]() {
+    return this.#uri;
+  }
+
+  get [kBunUndiciProxyAgentToken]() {
+    return this.#token;
+  }
+
+  // get [kBunUndiciProxyAgentHeaders]() {
```

Comment:
If think headers should be merged with the incoming headers, but if we can't maybe throw a notImplemented error/warning ?